content/blog/2023-remote-data-101/index.mdx

```diff
@@ -0,0 +1,145 @@
+---
+path: '/blog/2023-remote-data-101'
+date: '2023-06-11T11:12:41.253Z'
+title: 'Working with Remote Data 101'
+image: header.jpg
+author: Wojciech Ogrodowczyk
+---
+
+> God, grant me the serenity to accept the things I cannot change,<br/>
+> the courage to change the things I can,<br/>
+> and the wisdom to know the difference.
+
+You've probably heard [at least one of the versions of this Serenity prayer](https://en.wikipedia.org/wiki/Serenity_Prayer). It's not only a pretty useful piece of advice for life, but also a helpful tip towards handling remote data successfully when building mobile apps.
+
+One of the key strategies in building a reliable app is to understand what data is flowing through your system. Unless you work in a small team that believes in _full-stack development_, you won't have much control over what comes to your app from the backend services. This data can come with multiple types of issues with a varying impact:
+
+* It can simply be invalid for your app. Let's say you have a B2C ecommerce app and you get a product without a name or price. While (for various legacy reasons) this might be valid on the backend it definitely is not a product that you can display like all the others. Either you'll have to build a special version of all the product details screens and listings to accomodate a product like that, or you just ignore it.
+* Data can be malformed. An ID that is a number is passed as a string, or URLs for any detailed pages are not localised. Those are issues that can (and have to!) be corrected _somewhere_ in your app.
+* Cosmetic issues: typos in property names, mixing snake case and camel case, misleading naming (field called `distanceInMiles` that returns kilometers instead), etc. Those issues can be fixed on your side, or accepted as quirks and propagated across your codebase as well.
```

Comment:
I see in the meantime we've improved on how we render `inline code blocks` ❤️. It looks a bit off when you select this text with mouse, but overall to me its  much better now